src/core/ddsi/src/q_init.c

```diff
@@ -1906,7 +1903,8 @@ void rtps_fini (struct ddsi_domaingv *gv)
 
   xeventq_free (gv->xevents);
 
-  if (gv->config.xpack_send_async)
+  // if sendq thread is started
+  if (gv->sendq_running)
```

Comment:
There won't be any races at this point, but if a mutex is added to make setting `sendq_running` and starting the thread an atomic operation, I would probably lock it here, too. At least it'd pacify Coverity.